airflow/cli/commands/kubernetes_command.py

```diff
@@ -36,6 +37,12 @@
 from airflow.utils.cli import get_dag
 from airflow.utils.providers_configuration_loader import providers_configuration_loaded
 
+warnings.warn(
+    "Use kubernetes command from providers package, Use cncf.kubernetes provider >= 8.2.1",
+    DeprecationWarning,
+    stacklevel=2,
+)
```

Comment:
Does this mean that logic should be maintained in both files until deprecation? Is there a way to avoid duplication?

Reply:
Like @potiuk said in the description of the issue
> leaving the old implementation in core for old providers

Reply:
Yeah. We might decide in the future that we do not allow older provider versions (we can check it at runtime - similarly to what we propose to do for openlineage in https://github.com/apache/airflow/pull/39513#discussion_r1596005178 - and then we will be able to remove it from the core. But that is something maybe in a year or two.